pkg/util/filters.go

```diff
@@ -93,3 +93,27 @@ func PrepareFilters(r *http.Request) (*map[string][]string, error) {
 	}
 	return &filterMap, nil
 }
+
+// MatchLabelFilters matches labels and returs true if they are valid
+func MatchLabelFilters(filterValues []string, labels map[string]string) bool {
+	for _, filterValue := range filterValues {
+		matched := false
+		filterArray := strings.SplitN(filterValue, "=", 2)
+		filterKey := filterArray[0]
+		if len(filterArray) > 1 {
+			filterValue = filterArray[1]
+		} else {
+			filterValue = ""
+		}
+		for labelKey, labelValue := range labels {
+			if labelKey == filterKey && (filterValue == "" || labelValue == filterValue) {
+				matched = true
+				break
+			}
+		}
+		if !matched {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
I think we can make this function a bit cleaner, like this:
```golang
outer:
	for _, filterValue := range filterValues {
		filterArray := strings.SplitN(filterValue, "=", 2)
		filterKey := filterArray[0]
		if len(filterArray) > 1 {
			filterValue = filterArray[1]
		} else {
			filterValue = ""
		}
		for labelKey, labelValue := range labels {
			if labelKey == filterKey && (filterValue == "" || labelValue == filterValue) {
				continue outer
			}
		}
		return false
	}
return true
```

Reply:
I personally do not like continue outer formula, so I have chosen a version from containers, however, it is indifferent to me since both do the same.
Seems there are some CI problems and at least part is not related to this commit so I will continue once that gets sorted.